src/engine/v3/V3Interpreter.v3

```diff
@@ -1148,6 +1154,30 @@ component V3Interpreter {
 		var high = 0ul;
 		push(Value.V128(val, high));
 	}
+	def doLoadExtend<N, W, uW>(size: byte, log2_size: byte, view: u64 -> N, extend: N -> W, convert: W -> uW) {
+		var t = doLoadReg(8);
+		if (t.reason != TrapReason.NONE) return trap(t.reason);
+		var val = DataReaders.read_range_u64(t.result);
+		var low = 0ul, high = 0ul;
+		def half_lanes = 4 >> log2_size;
+		def src_shift = u6.view(size << 3);
+		def dst_shift = u6.view(src_shift << 1);
+		var shift: u6 = 0;
+		for (i < half_lanes) {
```

Comment:
Is it possible to do `for (i = 0; i < 64; i += size)` here?

Reply:
Yes. Thanks! Pushed a change.